src/util.cpp

```diff
@@ -334,7 +334,7 @@ static void lPrint(const char *type, bool isError, SourcePos p, const char *fmt,
     // exact same error message.  If so, return, so we don't redundantly
     // print it and annoy the user.
     static std::set<std::string> printed;
-    if (printed.find(formattedBuf) != printed.end()) {
+    if (isError && printed.find(formattedBuf) != printed.end()) {
```

Comment:
Why do need that?

We don't want to duplicate both errors and warnings.

Reply:
This routine is used for debug output. I think we do need debug messages whether they're duplicate or not.

Reply:
It's used for error, warnings (including performance warnings), and debug prints.

We definitely don't want duplication for errors and warnings. Debug prints should allow duplication.

The reason for duplicated errors and warnings is that we duplicate extern functions (masked and unmasked version) and they trigger the same messages, which is very annoying and confusing.

Reply:
Ok, I removed this change. It requires some extra efforts.